apps/studio/src/lib/editor/languageData.ts

```diff
@@ -126,16 +126,49 @@ export function getLanguageByName(name: string): LanguageData | undefined {
   return Languages.find((lang) => lang.name === name);
 }
 
-type Language = ConnectionType | "sql" | "text" | "html" | "json";
-
 interface CodeMirrorLanguage {
   mode: string | Record<string, unknown>;
   hint?: unknown;
 }
 
-export function resolveLanguage(lang: Language): CodeMirrorLanguage {
+export function resolveLanguage(lang: "sql", type: ConnectionType): CodeMirrorLanguage;
+export function resolveLanguage(lang: "text" | "html" | "json"): CodeMirrorLanguage;
+export function resolveLanguage(lang: "text" | "html" | "json" | "sql" , type?: ConnectionType): CodeMirrorLanguage {
```

Comment:
No let's get rid of this all together for sql dialects -- the issue is:

- We don't want to have to change this file when we add a new database type

The change you've made doesn't solve this, we still have `resolveSqlLanguage`. Instead, can we move this to the `DialectData` for each language, or someplace else we're editing anyway when we add a new language?